### PR TITLE
De 1101 tap hubspot seemingly fails to exit

### PR DESCRIPTION
**Pull request by wizhi**

### CODE REVIEW

File: tap_hubspot/hubspot.py [hunk 1]
```diff
@@ -406,41 +471,31 @@ def get_companies(
...
         for chunk in chunker(gen, 100):
-            ids: List[str] = [contact["id"] for contact in chunk]
-
-            companies_associations = self.get_associations(obj_type, "companies", ids)
-
-            for i, contact_id in enumerate(ids):
-                contact = chunk[i]
-
-                companies = companies_associations.get(contact_id, [])
+            ids: List[str] = [contact["id"] for contact, _ in chunk]
```
Comment (miquelpuigmena) [inline tap_hubspot/hubspot.py:486]: It seems like you try to unpack `chunk` to 2 vars. But `chunk` comes from `chunker(.)` that returns a list[T] (`def chunker(iter: Iterable[T], size: int) -> Iterable[List[T]]:`).

I was expecting `[contact["id"] for contact in chunk]` instead.

Comment (miquelpuigmena) [reply tap_hubspot/hubspot.py:486]: ah, gotcha, generic T infered from the generator 👍

**Review by JingLin0 — APPROVED** [review]

File: tap_hubspot/hubspot.py [hunk 2]
```diff
@@ -101,7 +163,10 @@ def streams(self, start_date: datetime, end_date: datetime):
         elif self.tap_stream_id == "companies":
             yield from self.get_companies_legacy()
         elif self.tap_stream_id == "contacts":
-            yield from self.get_contacts(start_date=start_date, end_date=end_date)
+            self.event_state["contacts_start_date"] = start_date
+            self.event_state["contacts_end_date"] = end_date
```
Comment (miquelpuigmena) [inline tap_hubspot/hubspot.py:167]: I don't see why this is important. Looks like it could be part of the exit condition. Is this a key part?

**Review by miquelpuigmena — APPROVED** [review]
The flow seems to work fine but I can't tell how this fix solves the bug you describe. I trust it does.

Also, I can't tell anything in regards to the fields that you set as mandatory (L62-L121).

**Pull request merged**